### ISSUE TITLE
Train-test mismatch

### ISSUE THREAD

**Issue by DanqiZhao21** [liyingyanUCAS/WoTE#27]
Hi, thanks for releasing the code!

I have a question about the trajectory offset branch and the reward/world-model branch. It seems that during training, the reward feature is extracted using the fixed WoTE anchors, while during evaluation/inference, the reward network uses the offseted anchors:

```python
if self.is_eval:
    offseted_trajectory_anchors = init_trajectory_anchor + trajectory_offset
    ego_feat_for_reward_network, _ = self.encode_traj_into_ego_feat(
        ego_status_feat, offseted_trajectory_anchors, batch_size
    )
else:
    ego_feat_for_reward_network = ego_feat_fixed_anchor_WoTE
```

Is this train/eval difference intentional? Is the reward model supervised with labels computed for the fixed anchors, which is why training uses fixed anchors for stability?

Also, if we want to train the reward/world-model branch with offseted anchors, would you recommend using `trajectory_offset.detach()` to prevent reward/world-model losses from updating the offset head?

Thanks!
